src/sentry/middleware/ratelimit.py

```diff
@@ -40,13 +41,21 @@ def __call__(self, request: Request) -> Response:
 
     def process_view(self, request: Request, view_func, view_args, view_kwargs) -> Response | None:
         """Check if the endpoint call will violate."""
+        rate_limit_config = None
+
         with metrics.timer("middleware.ratelimit.process_view"):
             try:
                 # TODO: put these fields into their own object
                 request.will_be_rate_limited = False
                 request.rate_limit_category = None
                 request.rate_limit_uid = uuid.uuid4().hex
-                request.rate_limit_key = get_rate_limit_key(view_func, request)
+                view_class = getattr(view_func, "view_class", None)
+                if view_class:
+                    rate_limit_config = get_rate_limit_config(view_class)
+                request.rate_limit_group = (
```

Comment:
I see you're putting the rate limit group on the request. Remember how we talked about making state that you attach to the global context purposeful, contained, and immutable. Does attaching the group to the request directly serve those heuristics? Also, if the config changes somehow or we wanted to attach different parts of it to the request, how would this code have to change?